bothub_nlp_nlu_worker/bothub_nlp_nlu/pipeline_components/preprocessing.py

```diff
@@ -0,0 +1,191 @@
+        ARE_WORD = " are"
+        NOT_WORD = " not"
+
+        # replace regex by MKT_WORD
+        phrase = re.sub(mkt_regex, MKT_WORD, phrase)
+        # replace regex by OK_WORD
+        phrase = re.sub(ok_regex, OK_WORD, phrase)
+        # replace regex by TY_WORD
+        phrase = re.sub(ty_regex, TY_WORD, phrase)
+        # replace regex by THX_WORD
+        phrase = re.sub(thx_regex, TY_WORD, phrase)
+        # replace regex by TKS_WORD
+        phrase = re.sub(tks_regex, TY_WORD, phrase)
+        # replace regex by AM_WORD
+        phrase = re.sub(am_regex, AM_WORD, phrase)
+        # replace regex by ARE_WORD
+        phrase = re.sub(are_regex, ARE_WORD, phrase)
+        # replace regex by NOT_WORD
+        phrase = re.sub(not_regex, NOT_WORD, phrase)
+
+        return phrase
+
+    @classmethod
+    def create(
+        cls, component_config: Dict[Text, Any], config: RasaNLUModelConfig
+    ) -> "Preprocessing":
+        # component_config = override_defaults(cls.defaults, component_config)
+        language = config.language
+        return cls(component_config, language)
+
+    def provide_context(self) -> Dict[Text, Any]:
+        return {'language': self.language}
+
+    def train(
+        self,
+        training_data: TrainingData,
+        config: Optional[RasaNLUModelConfig] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Train this component"""
+        not_repeated_phrases = set()
+        size = len(training_data.training_examples)
+        subtract_idx = 0
+
+        for idx in range(size):
+            example_text = training_data.training_examples[idx - subtract_idx].text
+            # removing accent and lowercasing characters
+            example_text = unidecode(example_text.lower())
+
+            if config.language == "en":
+                example_text = self.english_preprocesing(example_text)
+
+            if config.language == "pt_br":
+                example_text = self.portuguese_preprocesing(example_text)
+
+            if example_text in not_repeated_phrases:
+                # remove example at this index from training_examples
+                training_data.training_examples.pop(idx - subtract_idx)
+                subtract_idx += 1
+            else:
+                not_repeated_phrases.add(example_text)
+                print(example_text)
+                training_data.training_examples[idx - subtract_idx].text = example_text
+
+    def process(
+        self,
+        message: Message,
+        **kwargs: Any,
+    ) -> None:
+        """Process an incoming message."""
+        print(self.language)
+        APOSTROPHE_OPTIONS = ["'", "`"]
+
+        # removing accent and lowercasing characters
+        message.text = unidecode(message.text.lower())
+        # remove apostrophe from the phrase (important be first than s_regex regex)
+        for APOSTROPHE in APOSTROPHE_OPTIONS:
+            message.text = message.text.replace(APOSTROPHE, "")
+
+        if self.language == "en":
```

Comment:
I don't see this as a good practice, the ideal would be to use the Adapter structural design pattern